### PR TITLE
Remove confirm modals from the VBA flow

### PR DESCRIPTION
**Pull request by marcaaron**
### Fixed Issues
$ Expensify/Expensify#176726

### `bank-account/new` / Plaid flow 
1. Navigate to `/bank-account/new`
2. Select "Log into your bank"
3. Enter plaid test credentials (Bank: Chase, User: user_good, Pass: pass_good) **Note:** This will only work on native when on the staging secure server which we can switch on in `settings/preferences`.
4. When on this screen verify that you can press "Save & Continue" and the button is not disabled.
5. Press the button
6. Verify that a message appears above the button
7. Verify that the Choose an account field is highlighted in red

### `bank-account/new` / Add Manual flow
1. Navigate to `/bank-account/new`
2. Select "Connect manually"
3. Verify the "Save & Continue" button is not disabled
4. De-select the "I accept" checkbox so it is not checked
4. Press the "Save & Continue" button
5. Verify that a message appears above the button
6. Verify that the checkbox is outlined in red border
7. Enter plaid test account info (Routing: `011401533`, Account: `1111222233331111`) **Note:** This will only work on native when on the staging secure server which we can switch on in `settings/preferences`.
8. Check the checkbox
9. Verify the red border is cleared
9. Press "Save & Continue"
10. Verify we advance to the next step

### `bank-account/company`

1. Verify Save & Continue button is not disabled
2. Press the button
3. Verify all fields are highlighted in red (**Note:** Not all fields will have messages under them).
4. Enter information into all the fields 
5. Verify the errors are cleared from all fields
6. Press "Save & Continue"

### `bank-account/personal-information`

1. Verify the "Save & Continue" button is not disabled
2. Press the button
3. Verify the message appears above the submit button
4. Press the "fix the errors" link and verify the view scrolls to the top
4. Verify the Checkbox is highlighted and a message displayed under it.
5. Check the checkbox and verify the error clears
6. Sumbit the form again and verify that other missing fields are highlighted
7. Fill out all fields except the "State" and "City" fields
8. Submit the form
11. Verify error appears above subimit button + State selector is highlighted + text appears below the field
12. Select a state and verify the error clears
11. Submit again
9. Verify error text appears above the button + error text appears for the City field
10. Enter a city and verify the error clears 
11. Submit again

### `bank-account/contract`

1. Verify the "Save & Continue" button is not disabled
2. Check "I own more than 25% of..."
2. Press the button
3. Verify error appears above submit button + terms and conditions checkbox is highlighted + message displayed under
4. Check the box and verify the error clears
5. Submit again
6. Verify the "certify true and accurate" error appears + checkbox is highlighted.
7. Check the box and verify the error clears

### `bank-account/validate`

1. Create a new account on a non public domain
2. Create a workspace and navigate to it 
3. Press Get Started button
4. Move through the entire VBA flow to create an account in a Pending state that needs the 3 validation amounts
5. Enter the amounts wrong
6. Verify that a error appears above submit button instructing the user of the incorrect amounts
7. Enter the correct amounts

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [x] iOS
- [x] Android

#### iOS

680f2.png)

### CODE REVIEW

**Comment by marcaaron** [comment]
This is testing well for me on iOS so I'm going to take it off draft.

File: src/components/AddPlaidBankAccount.js [hunk 1]
```diff
@@ -173,40 +171,32 @@ class AddPlaidBankAccount extends React.Component {
...
-                                text={this.props.translate('common.saveAndContinue')}
-                                isLoading={this.state.isCreatingAccount}
-                                onPress={this.selectAccount}
+                    <ReimbursementAccountForm
+                        onSubmit={this.selectAccount}
+                    >
+                        {!_.isEmpty(this.props.text) && (
+                            <Text style={[styles.mb5]}>{this.props.text}</Text>
+                        )}
+                        {/* @TODO there are a bunch of logos to incorporate here to replace this name
```
Comment (marcaaron) [inline src/components/AddPlaidBankAccount.js:180]: Waiting on `#espanol` translation for this..

File: src/pages/ReimbursementAccount/ReimbursementAccountForm.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        let error = '';
+
+        if (!_.isEmpty(this.props.reimbursementAccount.errorModalMessage)) {
+            error = (
+                <Text style={styles.mutedTextLabel}>{this.props.reimbursementAccount.errorModalMessage}</Text>
+            );
+        } else {
+            error = (
+                <>
+                    {/* @TODO translate {this.props.translate('bankAccount.defaultFormAlertPrompt')} */}
```
Comment (marcaaron) [inline src/pages/ReimbursementAccount/ReimbursementAccountForm.js:48]: Waiting on #espanol translation for this.

Comment (marcaaron) [reply src/pages/ReimbursementAccount/ReimbursementAccountForm.js]: completo

**Review by ctkochan22 — DISMISSED** [review]

**Review by aldo-expensify — DISMISSED** [review]
Tested on web, code looks good to me!

**Comment by luacmartins** [comment]
Haven't looked at the code yet, but ran some tests.

I had issues with the `personal-information` step on web. Fields displayed errors seemingly out of order. 

https://user-images.githubusercontent.com/22219519/133703825-f1f9fa85-979c-423d-9e4b-74878b630743.mov

Also in the contract step, when pressing the `Save & continue` button without any boxes checked, I only got one error at a time instead of errors for both required checkboxes in one go, like in previous steps.

**Comment by marcaaron** [comment]
>I had issues with the personal-information step on web. Fields displayed errors seemingly out of order. Is this expected?

Yeah this is expected for now. We haven't yet updated these steps so they will show multiple errors at once.

>Also in the contract step, when pressing the Save & continue button without any boxes checked, I only got one error at a time instead of errors for both required checkboxes in one go, like in previous steps.

Same deal. We still have to fix those screens up, but will do it post launch.

**Review by luacmartins — COMMENTED** [review]
Let a small comment. Other than that, tests well on all platforms! Great work!

**Review by AndrewGable — DISMISSED** [review]

File: src/languages/en.js [hunk 3]
```diff
@@ -381,9 +381,6 @@ export default {
         checkHelpLine: 'Your routing number and account number can be found on a check for the account.',
         hasPhoneLoginError: 'To add a verified bank account please ensure your primary login is a valid email and try again. You can add your phone number as a secondary login.',
         hasBeenThrottledError: ({fromNow}) => `For security reasons, we're taking a break from bank account setup so you can double-check your company information. Please try again ${fromNow}. Sorry!`,
-        confirmModalTitle: 'Oops',
-        confirmModalPrompt: 'Please double check any highlighted fields and try again.',
-        confirmModalConfirmText: 'Got it',
```
[the remark this answers is not in the document]
Comment (marcaaron) [reply src/languages/en.js:386]: Damn! No it hasn't been removed. But even if we were planning to remove it's still referenced so this is a bug. Nice catch.

**Comment by marcaaron** [comment]
Thanks for catching that one @luacmartins. Updated.

**Review by luacmartins — DISMISSED** [review]
LGTM!

**Comment by marcaaron** [comment]
Fixed conflicts

**Review by luacmartins — APPROVED** [review]

**Review by AndrewGable — APPROVED** [review]

**Comment by isagoico** [comment]
We were able to experience the same behaviour in the Personal Information step here https://github.com/Expensify/App/pull/5311#issuecomment-921350746. Marking the PR as pass since it is expected behaviour.

**Pull request merged**